### PR TITLE
particle system for vulkan branch + enable/disable depth writing

### PR DESCRIPTION
**Pull request by sushantojal**
Porting the particle system to work with openGL on the vulkan branch. Doesnt yet work with vulkan.
Also adding changes which enable disable depth mask from the application.
Majority of the changes are to the shaders.

### CODE REVIEW

File: GVRf/Extensions/gvrf-particlesystem/build.gradle [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+        debug {
+            jniDebuggable true
+        }
+    }
+
+    sourceSets.main {
+        java.srcDirs = ['src/main/java']
+        //jni.srcDirs = [] // no auto generation of Android.mk
+        //jniLibs.srcDir 'src/main/jni/prebuilt' // pre-compiled libraries
```
Comment (dnvasilev) [inline GVRf/Extensions/gvrf-particlesystem/build.gradle]: Please don't check in commented out code in gradle files

File: GVRf/Extensions/gvrf-particlesystem/src/main/java/org/gearvrf/particlesystem/GVREmitter.java [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    private float[] BVSpawnTimes;
+    private float[] BVVelocities;
+
+    private float currTime = 0;
+
+    public GVREmitter(GVRContext gvrContext)
+    {
+        super(gvrContext);
+        mGVRContext = gvrContext;
+//        mGVRContext.registerDrawFrameListener(this);
```
Comment (dnvasilev) [inline GVRf/Extensions/gvrf-particlesystem/src/main/java/org/gearvrf/particlesystem/GVREmitter.java]: Please don't check in commented-out code without a good reason (good reason shall be explained in a detailed comment next to commented-out line)

**Comment by NolaDonato** [comment]
Is there anything in the particle system that would prevent it from working with Vulkan? If it does not use lighting or external textures we probably could get it to work.

**Comment by sushantojal** [comment]
Have made changes requested on the master branch as well as small modifications needed to make it work with vulkan.

**Comment by liaxim** [comment]
Needs DCO

**Comment by liaxim** [comment]
Demo works

**Comment by sushantojal** [comment]
GearVRf-DCO-1.0-Signed-off-by: Sushant Ojal
sushant.o@partner.samsung.com

**Comment by liaxim** [comment]
https://github.com/Samsung/GearVRf/pull/1277 has been merged. This can be considered for merging too.

File: GVRf/Framework/framework/src/main/jni/engine/renderer/gl_renderer.cpp [hunk 3]
```diff
@@ -414,6 +419,11 @@ namespace gvr
             GL(glEnable(GL_DEPTH_TEST));
         }
 
+        if (!render_data->depth_mask())
```
Comment (dnvasilev) [inline GVRf/Framework/framework/src/main/jni/engine/renderer/gl_renderer.cpp]: If depth mask is not set in RenderData, enable render mask? something wrong here.

Comment (dnvasilev) [inline GVRf/Framework/framework/src/main/jni/engine/renderer/gl_renderer.cpp]: Yep this is Restore and everything is in reverse

**Comment by dnvasilev** [comment]
Includes port of #1270

**Pull request merged**